### PR TITLE
Upload windows, mac, linux tarballs when a tag is pushed

### PR DESCRIPTION
**Pull request by bredelings**
This branch adds a new github action `release.yml` that is similar to `build.yml` except that it uploads the tarballs to a github draft release.  The draft release is invisible, but people in the revbayes org can see it.  So you can download the windows/mac/linux tarballs from it.  You can also edit it to either (i) delete it or (ii) add a summary and make it visible to everybody.

In preparation for this, the branch also modifies `build.yml` so that windows, mac, and linux all install revbayes to the same place -- ${HOME}/local .

### CODE REVIEW

File: projects/meson/README.md [hunk 1]
```diff
@@ -167,7 +167,23 @@ ninja -C build-gtk install
...
+    * The prefix `x86_64-w64-` indicates that we want to compile for 64-bit windows, not 32-bit windows.
+    * The ending `-posix` indicates that we want to use UNIX (POSIX) threads, not Windows threads.
+
+    There is also a homebrew package `mingw-w64` on Mac, but these instructions have not been tested on Mac.
+
+        brew install mingw-w64 wine-stable
+
+    You might need to update the paths in the cross-file below.
+
+2. Create a fake windows root directory and download windows libraries from MINGW64.
```
Comment (davidcerny) [inline projects/meson/README.md:186]: Could it be worth providing an example command showing how to do this?

**Review by davidcerny — COMMENTED** [review]
All the code changes generally look good to me, and the more detailed meson instructions seem helpful. Unfortunately, I wasn't able to test the Windows-related changes, since I don't have access to any machine running that OS.

File: projects/meson/README.md (hunk 1 above)
Comment (bredelings) [reply projects/meson/README.md:186]: Do you mean, to update the paths?

Comment (davidcerny) [reply projects/meson/README.md:184]: I did mean creating a fake windows root directory, but somehow I failed to notice that the command immediately below the line in question did exactly that! Approving now.

**Review by davidcerny — APPROVED** [review]

**Pull request merged**